File: e2e-tests/spring-rest-openapi-v2/src/test/java/com/foo/rest/examples/spring/db/existingdata/ExistingDataController.java
package com.foo.rest.examples.spring.db.existingdata;

import com.foo.rest.examples.spring.db.SpringWithDbController;
import org.evomaster.client.java.controller.internal.db.DbSpecification;

/**
 * Created by arcuri82 on 19-Jun-19.
 */
public class ExistingDataController extends SpringWithDbController {

    public ExistingDataController() {
        super(ExistingDataApp.class);
    }

    @Override
    public void resetStateOfSUT() {
        super.resetStateOfSUT();

//        ExistingDataEntityX x = new ExistingDataEntityX();
//        x.setId(42L);
//        x.setName("Foo");
//
//        ExistingDataRepositoryX rep = ctx.getBean(ExistingDataRepositoryX.class);
//        rep.save(x);
    }

    @Override
    public DbSpecification setDbSpecification() {
        DbSpecification spec =  super.setDbSpecification();
        if (spec != null)
            spec.initSqlScript= "INSERT INTO EXISTING_DATA_ENTITYX (ID, NAME) VALUES (42, 'Foo')";
        return spec;
    }
}

File: e2e-tests/spring-graphql/src/test/kotlin/com/foo/graphql/db/exisitingdata/ExistingDataController.kt
package com.foo.graphql.db.exisitingdata

import com.foo.graphql.db.SpringWithDbController
import org.evomaster.client.java.controller.internal.db.DbSpecification


class ExistingDataController : SpringWithDbController(ExistingDataApplication::class.java) {

    override fun schemaName() = "dbexisting.graphqls"


    override fun resetStateOfSUT() {
        super.resetStateOfSUT()

//        val rep = ctx!!.getBean<ExistingDataXRepository>(ExistingDataXRepository::class.java)
//        rep.save(ExistingDataX(42L, "Foo"))
    }

    override fun getDbSpecification(): DbSpecification? {
        val spec = super.getDbSpecification();
        spec?.initSqlScript = "INSERT INTO EXISTING_DATAX (ID, NAME) VALUES (42, 'Foo')";
        return spec
    }
}